src/Lykke.Service.BlockchainHistoryReader.AzureRepositories/Implementations/HistorySourceLockRepository.cs

```diff
@@ -0,0 +1,159 @@
+using System;
+using System.Threading.Tasks;
+using Common.Log;
+using JetBrains.Annotations;
+using Lykke.Common.Log;
+using Lykke.SettingsReader;
+using Microsoft.AspNetCore.Http;
+using Microsoft.WindowsAzure.Storage;
+using Microsoft.WindowsAzure.Storage.Blob;
+
+
+namespace Lykke.Service.BlockchainHistoryReader.AzureRepositories.Implementations
+{
+    [UsedImplicitly]
+    public class HistorySourceLockRepository : RepositoryBase, IHistorySourceLockRepository
+    {
+        private static readonly TimeSpan LockDuration = TimeSpan.FromSeconds(60); 
+        
+        private readonly CloudBlobContainer _container;
```

Comment:
Why Lykke.AzureStorage is not used?

Reply:
Lykke.AzureStorage does not provide us necessary functionality. Should I move it to the common library?